### PR TITLE
rfc0023: mark `did` as required

### PR DESCRIPTION
**Pull request by llorllale**
Part of response to issue in AFGO: https://github.com/hyperledger/aries-framework-go/issues/2415

Signed-off-by: George Aristy <george.aristy@securekey.com>

### CODE REVIEW

**Comment by llorllale** [comment]
Note that `did` MUST be present due to the self-certifying nature of some "ledger-less" DIDs (eg. `did:peer`).

**Comment by llorllale** [comment]
Is there any reason why `did` should be optional?

**Review by TelegramSam — APPROVED** [review]

File: features/0023-did-exchange/README.md [hunk 1]
```diff
@@ -321,8 +321,8 @@ The key used in the signed attachment must be verified against the invitation's
 #### Response Message Attributes
 
 * The `@type` attribute is a required string value that denotes that the received message is an exchange request.
-* The `~thread` block contains a `thid` reference to the `@id` of the request message.
-* The `did` attribute is a required string value and denotes DID in use by the responder. Note that this MAY NOT be the same DID used in the invitation.
+* The `~thread` decorator MUST be included. It contains a `thid` reference to the `@id` of the request message.
+* The `did` attribute MUST be included. It denotes the DID in use by the responder. Note that this MAY NOT be the same DID used in the invitation.
```
Comment (andrewwhitehead) [inline features/0023-did-exchange/README.md:325]: Can we rephrase to remove the 'MAY NOT'? That's not actually RFC2119 verbiage.

Comment (llorllale) [reply features/0023-did-exchange/README.md:325]: Discussed on today's B call:

The "MAY NOT" verbiage was already present in the RFC and not altered in any meaningful way by this PR. The purpose of this PR is to mark `did` as required.

**Comment by llorllale** [comment]
@dhh1128 branch updated

**Comment by dhh1128** [comment]
approved on community B call 24 Mar 2021

**Pull request merged**